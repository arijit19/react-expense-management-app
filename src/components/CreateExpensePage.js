import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {AddExpense} from '../Redux/Actions/ExpenseActions.js'

export class CreateExpense extends React.Component {
    onSubmit = (expense)=>{
        this.props.addExpense(expense);
        this.props.history.push('/');
    }

    render(){
        return (
            <div>
                <p>This is add expense page</p>
                <ExpenseForm onSubmit={this.onSubmit}/>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch)=> ({
    addExpense: (expense)=> dispatch(AddExpense(expense))
})
export default connect(undefined, mapDispatchToProps)(CreateExpense);