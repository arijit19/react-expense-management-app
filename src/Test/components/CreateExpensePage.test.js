import React from 'react';
import {shallow} from 'enzyme';

import {CreateExpense} from '../../components/CreateExpensePage.js'
import expenses from '../fixtures/Expense.js'

let AddExpenseSpy, historySpy, wrapper;

beforeEach(()=>{
    AddExpenseSpy = jest.fn();
     historySpy = {push: jest.fn()}
     wrapper = shallow(<CreateExpense addExpense={AddExpenseSpy} history={historySpy}/>);
});

test('Testing Adding Expences component render', ()=>{   
    expect(wrapper).toMatchSnapshot();
});

test('Testing Adding Expences component when form is submited', ()=>{
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[0]);
    expect(historySpy.push).toHaveBeenLastCalledWith('/');
    expect(AddExpenseSpy).toHaveBeenLastCalledWith(expenses[0]);
});