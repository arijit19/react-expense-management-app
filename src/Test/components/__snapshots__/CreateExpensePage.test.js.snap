// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Adding Expences component render 1`] = `
<div>
  <p>
    This is add expense page
  </p>
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
